waltz-tools/src/main/java/com/wepay/waltz/tools/cluster/ClusterCli.java

```diff
@@ -287,6 +278,154 @@ protected void processCmd(CommandLine cmd) throws SubCommandFailedException {
+                        ValidationResult.ValidationType.PARTITION_ASSIGNMENT_ZK_SERVER_CONSISTENCY,
+                        status, error);
+                    partitionValidationResults.validationResultsMap.put(validationResult.type, validationResult);
+                }).exceptionally(e -> {
+                    ValidationResult validationResult = new ValidationResult(
+                        ValidationResult.ValidationType.PARTITION_ASSIGNMENT_ZK_SERVER_CONSISTENCY,
+                        ValidationResult.Status.FAILURE,
+                        e.getMessage());
+                    partitionValidationResults.validationResultsMap.put(validationResult.type, validationResult);
+                    return null;
+                });
+            } catch (Exception ex) {
+                return new CompletableFuture<>();
+            }
+        }
+
+        private CompletableFuture<Void> buildServerStorageConnectivityValidationPerPartition(
+            InternalRpcClient rpcClient, Endpoint serverEndpoint, ReplicaAssignments replicaAssignments,
+            int partitionId, PartitionValidationResults partitionValidationResults) throws InterruptedException {
+
+            try {
+                CompletableFuture<Map<Endpoint, Map<String, Boolean>>> connectivityStatusFuture =
+                    (CompletableFuture<Map<Endpoint, Map<String, Boolean>>>)
+                        rpcClient.checkServerConnections(Collections.singleton(serverEndpoint));
+
+                return connectivityStatusFuture.thenAccept(response -> {
+                    Map<Integer, Set<String>> partitionIdToReplicas = getPartitionToReplicasInfo(replicaAssignments);
+                    ValidationResult validationResult =
+                        buildStorageConnectivityValidationResult(partitionIdToReplicas.get(partitionId),
+                            response.get(serverEndpoint), String.format("Server %s", serverEndpoint));
+                    partitionValidationResults.validationResultsMap.put(validationResult.type, validationResult);
+                });
+            } catch (Exception ex) {
+                return new CompletableFuture<>();
+            }
+        }
+
+        private void updateRemainingValidationResults(CompletableFuture<Void> consistencyValidationFuture,
+                                                      CompletableFuture<Void> connectivityValidationFuture,
+                                                      List<PartitionValidationResults> partitionsValidationResultList) throws Exception {
+
+            // Update validation result for remaining partitions if not already updated.
+            String error = "";
+
+            try {
+                consistencyValidationFuture.get(TIMEOUT_IN_SECONDS, TimeUnit.SECONDS);
+            } catch (ExecutionException | TimeoutException exception) {
+                error = (exception.getMessage() == null) ? "No Server Endpoint found." : exception.getMessage();
```

Comment:
Actually my bad, I forgot that in `all partitions` case, the consistencyValidationFuture will be a completed future. Whereas in `per partition` case, I am returning a future which is not complete. 

Going to modify the code in `per partition` case to return `CompletableFuture.completedFuture(null)` so that both `all partition` and `per partition` implementation remains consistent. 

Also going to move the `No Server Endpoint found` error message as the default message.

Reply:
Will modify the same for the `connectivityValidationFuture` as well.